### PR TITLE
Merge Features/quick info party into master-vs-deps

### PR DESCRIPTION
**Pull request by JieCarolHu**
<details><summary>Merge Features/quick info party into master-vs-deps </summary>

### Customer scenario
The changes are already reviewed in the following PRs
https://github.com/dotnet/roslyn/pull/20554
https://github.com/dotnet/roslyn/pull/25181
https://github.com/dotnet/roslyn/pull/25484
https://github.com/dotnet/roslyn/pull/25975

### Bugs this fixes
https://github.com/dotnet/roslyn/issues/24094
https://github.com/dotnet/roslyn/issues/24956
https://github.com/dotnet/roslyn/issues/24434
https://github.com/dotnet/roslyn/issues/25608

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

**Comment by JieCarolHu** [comment]
retest windows_debug_unit64_prtest please

**Comment by JieCarolHu** [comment]
retest windows_release_unit64_prtest please

File: README.md [hunk 1]
```diff
@@ -7,17 +7,15 @@
...
...ob/dotnet_roslyn/job/dev15.3.x/job/windows_debug_unit32/)|[![Build Status](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_debug_unit64/badge/icon)](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_debug_unit64/)|[![Build Status](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_release_unit32/badge/icon)](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_release_unit32/)|[![Build Status](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_release_unit64/badge/icon)](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_release_unit64/)|[![Build Status](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_determinism/badge/icon)](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_determinism/)|[![Build Status](https://ci.dot.net/buildStatus/icon?job=dotnet_roslyn/dev15.3.x/windows_debug_vs-integration)](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_debug_vs-integration/)|[![Build Status](https://ci.dot.net/buildStatus/icon?job=dotnet_roslyn/dev15.3.x/windows_release_vs-integration)](https://ci.dot.net/job/dotnet_roslyn/job/dev15.3.x/job/windows_release_vs-integration/)|
```
Comment (DustinCampbell) [inline README.md]: This looks very suspicious. :smile:

Comment (JieCarolHu) [reply README.md]: I probably resolved the merge conflict wrong, let me revert it

**Comment by JieCarolHu** [comment]
@jasonmalinowski Could you please approve this change? Thanks

**Comment by jasonmalinowski** [comment]
Also FYI to @Pilchie I think since we'll kinda want ask mode approval on this.

**Comment by KirillOsenkov** [comment]
I didn't review the changes but I'm guessing this is fine for VSMac. We are planning to adopt the new Quick Info API in 15.8.

File: src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs [hunk 2]
```diff
@@ -489,7 +489,6 @@ private static Document GetDocumentToVerify(DocumentId expectedChangedDocumentId
             int index, ImmutableArray<CodeAction> actions, CodeActionPriority? priority = null)
         {
             Assert.NotNull(actions);
-            Assert.NotEmpty(actions);
```
Comment (DustinCampbell) [inline src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs]: What is this change and how is it related to Quick Info work?

Comment (JieCarolHu) [reply src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs]: reverted. the change was accidentally got in when I was switching branches...

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.preview2

**Comment by DustinCampbell** [comment]
w00t!!!

**Pull request merged**